./cudf.h:24:10: fatal error: 'vector' file not found

I fail to build `ocaml-mccs` on OSX. When running `jbuilder build -p mccs`, there are many errors like the following:

```
          cc src/changed_criteria.o (exit 1)
(cd _build/default/src && …/bin/cc -I …/lib/ocaml -O2 -fno-strict-aliasing -fwrapv -D_FILE_OFFSET_BITS=64 -D_REENTRANT -I …/lib/ocaml/4.05.0/site-lib/bytes -I …/lib/ocaml/4.05.0/site-lib/extlib -I …/lib/ocaml/4.05.0/site-lib/cudf -I glpk -g -I . -DUSEGLPK -o changed_criteria.o -c changed_criteria.cpp)
clang-5.0: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
In file included from changed_criteria.cpp:9:
In file included from ./changed_criteria.h:12:
In file included from ./abstract_criteria.h:13:
./cudf.h:24:10: fatal error: 'vector' file not found
#include <vector>
         ^~~~~~~~
1 error generated.
```

Apparently, a C compiler is run to deal with C++ code.

How can I tell the build system what C++ compiler to use?

--- a/src/context_flags.ml
+++ b/src/context_flags.ml
@@ -24,6 +24,7 @@ let ifc c x = if c then x else []
 
 let cxxflags =
   let flags =
+    (ifc (Config.system = "macosx") ["-x"; "c++"]) @
     (ifc (Sys.win32 && Config.ccomp_type = "msvc") ["/EHsc"]) @
     (ifc useGLPK ["-DUSEGLPK"]) @
     (ifc useCOIN ["-DUSECOIN"]) @

